src/Workspaces/Remote/ServiceHub/Shared/RemoteEndPoint.cs

```diff
@@ -126,9 +129,45 @@ public async Task<T> InvokeAsync<T>(string targetName, IReadOnlyList<object> arg
 
             try
             {
-                await Extensions.InvokeAsync(_rpc, targetName, arguments, funcWithDirectStreamAsync, cancellationToken).ConfigureAwait(false);
+                using var mergedCancellation = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                using var stream = new ServerDirectStream();
+
+                Task? task = null;
+                try
+                {
+                    // send request by adding direct stream name to end of arguments
+                    task = _rpc.InvokeWithCancellationAsync(targetName, arguments.Concat(stream.Name).ToArray(), cancellationToken);
+
+                    // if invoke throws an exception, make sure we raise cancellation.
+                    RaiseCancellationIfInvokeFailed(task, mergedCancellation, cancellationToken);
```

Comment:
seems so strane to do it this way.

Reply:
IDK why is it done this way, but don't want to mess with this too much.